### PR TITLE
Fix coverity vulnerabilities 2

### PR DESCRIPTION
**Pull request by as0ler**
**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

Fix 1380409 Resource leak
Fix 1355077 Resource leak
Fix 1355075 Resource leak
Fix 1408978 Resource leak
Fix 1408693 Resource leak
Fix 1408947 Resource leak
Fix 1355076 Resource leak
Fix 1259928 Resource leak
Fix 1134937 Resource leak
Fix 1134819 Resource leak
Fix 1430478 Out-of-bounds read
Fix 1408737 Out-of-bounds write

### CODE REVIEW

File: libr/core/canal.c [hunk 1]
```diff
@@ -1880,12 +1886,12 @@ static int core_anal_graph_nodes(RCore *core, RAnalFunction *fcn, int opts, PJ *
...
 		pj_end (pj);
 	}
-	free (pal_jump);
-	free (pal_fail);
-	free (pal_trfa);
-	free (pal_curr);
-	free (pal_traced);
-	free (pal_box4);
+	R_FREE (pal_jump);
+	R_FREE (pal_fail);
```
Comment (trufae) [inline libr/core/canal.c:1890]: No need to R_FREE before return

File: libr/core/canal.c [hunk 2]
```diff
@@ -1831,6 +1831,12 @@ static int core_anal_graph_nodes(RCore *core, RAnalFunction *fcn, int opts, PJ *
 	char *pal_box4 = palColorFor ("graph.box4");
 	if (!fcn || !fcn->bbs) {
 		eprintf ("No fcn\n");
+		R_FREE (pal_jump);
```
Comment (trufae) [inline libr/core/canal.c:1834]: Just call free()

File: libr/core/canal.c (hunk 1 above)
Comment (as0ler) [reply libr/core/canal.c:1890]: If they are not freed, Coverity triggers an Resource leak issue.

File: libr/core/canal.c (hunk 2 above)
Comment (trufae) [reply libr/core/canal.c:1834]: R_FREE is free+ set Null. So if this gows before the return. Its extra work for the compiler to remove dead code

**Pull request merged**